spec/controllers/requests_controller_spec.rb

```diff
@@ -0,0 +1,46 @@
+require "rails_helper"
+include Pagy::Backend
+
+RSpec.describe RequestsController, type: :controller do
+  let!(:company) {FactoryBot.create :company}
+  let!(:office) {FactoryBot.create :office, company: company}
+  let!(:user) {FactoryBot.create :user, office: office, last_name: "!23"}
+  let!(:device) {FactoryBot.create :device, office: office}
+  let!(:request) {FactoryBot.create :request, device: device, user: user}
+
+  before{sign_in(user)}
+
+  describe "GET requests#index" do
+    context "success when valid attributes" do
+      before{get :index}
+
+      it "responds successfully" do
+        expect(response).to	have_http_status :ok
+      end
+
+      it "display the requests of the current user" do
+        expect(assigns(:requests)).to eq user.requests
+      end
+
+      it "show latest requests by creation date" do
+        expect(assigns(:requests).recent).to eq user.requests.recent
+      end
+
+      it "is expected to render template matcher index" do
+        expect(response).to render_template(:index)
+      end
+
+      it "no requests list" do
+        expect(assigns(:requests).blank?).to eq false
+      end
+
+      it "have data in requests list" do
+        expect(assigns(:requests).present?).to eq true
+      end
+
+      it "number of records in one page" do
+        expect(assigns(:pagy).items).to eq Settings.pagy.config.page.default
+      end
```

Comment:
Từ 14 đén 43 chỉ là 1 testcase thui nham các cái it là các mong đợi của test case đó? 1 trang như vậy mà chỉ có 1 test case hả?